### DISCUSSION TITLE
Allow clicking/picking without changing selection?

### DISCUSSION THREAD

**Discussion by ilyvion** [aevyrie/bevy_mod_picking#235] [Q&A]
I'm making a game where various things need to be selectable and clickable/pickable at various times. I've discovered that I can disable the pickable nature of a mesh by removing the `Pickable` component. I was hoping that I could remove the ability to *select* a mesh by removing the `PickSelection` component, but this doesn't seem to be a thing.

My logic was that since the `PickableBundle` adds the `PickSelection` component only if the `selection` feature is enabled, surely removing the component would remove the selection mechanism? Alas, no.

Before I get myself stuck in an XY problem, this is what I'm trying to accomplish: I have a board game, and on a player's turn, they are able to select their pieces, and once they select a piece, I show "target" meshes on places they can move their piece to. I want these target meshes to be *clickable*/*pickable* in `bevy_mod_picking`, but **not** *selectable*. The idea is that I want to be able to react to the player choosing a target, *but that mesh should not become selected* (it's about to immediately get hidden anyway) and, maybe more importantly, *the piece mesh should not become deselected* by choosing the target!

Any suggestions on how I can/should go about this would be greatly appreciated. I'll also take this opportunity to thank aevyrie and all contributors for this awesome Bevy plugin that made adding picking to my game an absolute breeze. ♥

**Comment by ilyvion** [comment]
I should say, removing `PickSelection` *partially* works, in that the target doesn't end up selected, but the piece gets *deselected* nonetheless, which is unlike when I click on nothing (I have `click_nothing_deselect_all` enabled) -- if only clicking on something with `PickSelection` missing didn't trigger a deselect just like clicking on nothing does, I'd have what I want, to be honest.

**Reply by aevyrie** [reply]
Have you tried the [`NoDeselect`](https://docs.rs/bevy_picking_selection/0.15.0/bevy_picking_selection/struct.NoDeselect.html) component? If you add this to an entity, it will not trigger a deselection when it is clicked. Please try out the deselection example to see if it meets your needs: https://github.com/aevyrie/bevy_mod_picking/blob/main/examples/deselection.rs

From your description, I think you just need to add the `NoDeselect` component to your "target" meshes.
